### ISSUE TITLE
Add indicator's short_name to catalog path to make it unique

### ISSUE THREAD

**Issue by Marigold** [owid/etl#1823]
Even though we still use variable ids, it might be interesting to start experimenting with paths. They might be helpful for our work on chart sync from staging servers, among other things.

Currently, we store `catalogPath` column in the variables table in the form
```
channel/namespace/version/dataset/table
```
the problem is that it doesn't reference variable's short name and is therefore not unique. We store `shortName` in the table too, so we could easily concat them, but I still think it'd be nice to have it all in a single column.

My favourite format is
```
channel/namespace/version/dataset/table#indicator_short_name
```
but I'm open to ideas.

## Technical implementation

* Update ETL to store the new format
* Migration that would append `#indicator_short_name` to catalogPath
* Add unique constraint in MySQL

**Comment by larsyencken** [comment]
I'm totally in favour. I've used `channel/namespace/version/dataset/table@indicator_short_name` in the past, but there's no difference between `#` and `@` really, so I have no strong preference.

We currently have a URI scheme, so we could also consider extending that:

`data://channel/namespace/version/dataset/table?col=indicator_short_name`

That could make sense if we imagined other arguments landing there, or if reusing URL parsers would make our lives easier.

One thing I'm acutely aware of is that each indicator is not actually a column, but a table with columns like `dim1,dim2,...,short_name`. So maybe the use of `col` in the URI scheme above is misleading.

**Comment by larsyencken** [comment]
> Yeah I think we should reserve the query param style for dimensions or other slicing operations in the future. `#short_name` sounds good to me. Are `short_name`'s guaranteed to be something like `[A-Za-z0-9_-]+`?

Short name should match `[a-z][a-z0-9_]+`, i.e. they should be snake case to make it easy to work with them in Pandas. I'm not sure where it's enforced.

When I check to make sure, I see around 200k indicators in the ETL at the moment, all snake case. About ~200 are actually year name columns prefixed with `_`, like `_2019` (e.g. `garden/excess_mortality/latest/wmd/wmd/_2023`) where the year columns should have been melted but it didn't happen.

<img width="1354" alt="Screenshot 2023-10-20 at 09 35 48" src="https://github.com/owid/etl/assets/36012/8cb681b5-b464-4239-96c6-44d000b2e28a">

Long story short though, they're all compatible with a URL scheme.

**Comment by Marigold** [comment]
@pabloarosado want to chime in? I don't have an opinion, I'm okay with both URI and just adding `#indicator` (though URI is slightly more appealing).

> Yeah I think we should reserve the query param style for dimensions or other slicing operations in the future.

@danyx23 I don't get it - why does it imply we shouldn't use URI format?

**Comment by pabloarosado** [comment]
I don't have a strong opinion on this. If the goal is to find a unique identifier of the variables, the simplest would be:
`namespace/version/dataset/table/short_name`. Channel is not necessary, since by construction it's always `grapher`, isn't it?
